### PR TITLE
Make Some update.

### PR DESCRIPTION
**Pull request by amreisa**
1 : Make Lib Move to C++11.
2 :  Re-Move QDebug code from release code.

### CODE REVIEW

File: src/tableset.h [hunk 1]
```diff
@@ -39,8 +39,8 @@ template<class T>
 class NUT_EXPORT TableSet : public TableSetBase
 {
 public:
-    explicit TableSet(Database *parent = nullptr);
```
Comment (HamedMasafi) [inline src/tableset.h]: Why optional parameter of constructor removed?

File: src/tableset.h [hunk 2]
```diff
@@ -39,8 +39,8 @@ template<class T>
...
-    explicit TableSet(Table *parent = nullptr);
```
Comment (HamedMasafi) [inline src/tableset.h]: Also here... ?

File: src/phrase.cpp [hunk 3]
```diff
@@ -20,11 +20,15 @@
...
 #include "phrase.h"
 
+#ifdef QT_DEBUG
 #include <QDebug>
+#endif // QT_DEBUG
 
 NUT_BEGIN_NAMESPACE
 
+#ifdef QT_DEBUG
 #define LOG(s) qDebug() << __func__ << s
```
Comment (HamedMasafi) [inline src/phrase.cpp]: These will be removed

File: src/database.cpp [hunk 4]
```diff
@@ -133,17 +137,23 @@ bool DatabasePrivate::updateDatabase()
     DatabaseModel current = currentModel;
 
     if (last == current) {
+#ifdef QT_DEBUG
         qDebug("Databse is up-to-date");
```
Comment (HamedMasafi) [inline src/database.cpp]: Some messages (like that is an information) and should be shown even in release mode. Qt itself contains lot of this messages. Like QObject::connect on invalid signal/slot. 
Some others is for debug purpose and should be remove completly

File: src/tableset.h (hunk 1 above)
Comment (amreisa) [reply src/tableset.h]: MSVC not like this I don't know Why !!!

File: src/database.cpp (hunk 4 above)
Comment (amreisa) [reply src/database.cpp]: Qdebug not run in release mode. That what I know.

**Comment by amreisa** [comment]
And if the porject use C++11 then we have nullptr and then we not need Q_NULLPTR.

File: src/database.cpp (hunk 4 above)
Comment (HamedMasafi) [reply src/database.cpp]: No QDebug run on release mode too unless QT_NO_DEBUG macro is defined

**Comment by amreisa** [comment]
It's good point. But you add C++11 in QMake, For that I edit code the code with this style.

**Pull request closed without merging**